### ISSUE TITLE
infer quant --ref and --index

### ISSUE THREAD

**Issue by kfuku52** [kfuku52/amalgkit#28]
As reference fasta should be identifiable from the input metadata table (by species name), we should support `quant --ref "infer"`,  which automatically detects the reference fasta in the default reference directory (I think we don't have such dir yet, so let's say `amalgkit_out/reference/`). File names should be like `Arabidopsis_thaliana_ANYANNOTATIONHERE.fasta.gz`. `quant --index "infer"` should also be supported similarly.

**Comment by Hego-CCTB** [comment]
Yeah, I thought about something like this. This would also tie in toa `amalgkit sanity` functionality we talked about in another thread.

**Comment by kfuku52** [comment]
Will you work on this?

**Comment by Hego-CCTB** [comment]
Sure! I actually already have code for something very close to this.

**Comment by Hego-CCTB** [comment]
https://github.com/kfuku52/amalgkit/issues/33
and 
https://github.com/kfuku52/amalgkit/issues/72
